iothub_client/src/iothubtransport_amqp_common.c

```diff
@@ -1146,18 +1147,17 @@ static int IoTHubTransport_AMQP_Common_Device_DoWork(AMQP_TRANSPORT_DEVICE_INSTA
         }
         else // i.e., DEVICE_STATE_ERROR_AUTH || DEVICE_STATE_ERROR_AUTH_TIMEOUT || DEVICE_STATE_ERROR_MSG
         {
-            LogError("Failed performing DoWork for device '%s' (device reported state %d; number of previous failures: %lu)",
-                STRING_c_str(registered_device->device_id), (int)registered_device->device_state, (unsigned long)registered_device->number_of_previous_failures);
-
             registered_device->number_of_previous_failures++;
 
             // Codes_SRS_IOTHUBTRANSPORT_AMQP_COMMON_09_046: [If the device has failed for MAX_NUMBER_OF_DEVICE_FAILURES in a row, it shall trigger a connection retry on the transport]
             if (registered_device->number_of_previous_failures >= MAX_NUMBER_OF_DEVICE_FAILURES)
             {
+                LogError("Failed performing DoWork for device '%s' (device reported state %d; number of previous failures: %lu)",
+                    STRING_c_str(registered_device->device_id), (int)registered_device->device_state, (unsigned long)registered_device->number_of_previous_failures);
                 result = MU_FAILURE;
             }
             // Codes_SRS_IOTHUBTRANSPORT_AMQP_COMMON_09_045: [If the registered device has a failure, it shall be stopped using amqp_device_stop()]
-            else if (amqp_device_stop(registered_device->device_handle) != RESULT_OK)
+            else if (amqp_device_delayed_stop(registered_device->device_handle, DEFAULT_DEVICE_STOP_DELAY) != RESULT_OK)
```

Comment:
DEFAULT_DEVICE_STOP_DELAY  
Timeout? 

```suggestion
            else if (amqp_device_delayed_stop(registered_device->device_handle, DEFAULT_DEVICE_STOP_TIMEOUT_SEC) != RESULT_OK)
```

Reply:
This period of time is really a delay, not a timeout.